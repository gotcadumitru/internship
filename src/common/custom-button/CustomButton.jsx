import React from 'react';
import './CustomButton.css';

const CustomButton = ({children, profilebtn,yellowbtn, whitebtn,searchbtn,...otherProps})=>{
    return (
        <button data-testid="submit-button" className={`customButton ${profilebtn ? "profilebtn" : ''} ${whitebtn ? "whitebtn" : ''} ${yellowbtn ? "yellowbtn" : ''} ${searchbtn ? "searchbtn" : ''}`}{...otherProps}>
           <span>
           {children }

           </span> 
        </button>
    );
}
export default CustomButton;







