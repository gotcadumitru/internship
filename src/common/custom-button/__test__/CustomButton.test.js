import { getByTestId, render, screen } from "@testing-library/react";
import CustomButton from '../CustomButton';
import renderer from 'react-test-renderer';
describe("custom-button",()=>{

    test("check the text inside the button",()=>{

        render(<CustomButton>Hello</CustomButton>)
        const helloElement = screen.getByText("Hello",{exact: false});
        expect(helloElement).toBeInTheDocument();
        
    })

    test("check class",()=>{

        const {container} = render(<CustomButton profilebtn searchbtn whitebtn>Hello</CustomButton>);        
        expect(container.firstChild).toHaveClass('profilebtn customButton whitebtn searchbtn',{exact: true});
        
    })
    test("check click",()=>{

        const {getByTestId} =  render(<CustomButton >Hello</CustomButton>);
        const button = getByTestId('submit-button');
        expect(button).toHaveTextContent("Hello");
    })
    test("create snapshot",()=>{

        const tree =  renderer.create(<CustomButton >Hello</CustomButton>).toJSON();
        expect(tree).toMatchSnapshot()
    })

    test("create snapshot with new class",()=>{

        const tree =  renderer.create(<CustomButton yellowbtn >Hello</CustomButton>).toJSON();
        expect(tree).toMatchSnapshot()
    })
    // test("check style",()=>{

    //     const {getByTestId} =  render(<CustomButton >Hello</CustomButton>);
    //     const button = getByTestId('submit-button');
    //     expect(button).toHaveStyle({"box-shadow": "0px 7px 14px 2px rgba(65, 73, 185, 0.2)"},{exact: false});
    // })
    
})