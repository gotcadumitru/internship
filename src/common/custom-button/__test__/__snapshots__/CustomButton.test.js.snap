// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-button create snapshot 1`] = `
<button
  className="customButton    "
  data-testid="submit-button"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`custom-button create snapshot with new class 1`] = `
<button
  className="customButton   yellowbtn "
  data-testid="submit-button"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`custom-button snapshot 1`] = `
<button
  className="customButton    "
  data-testid="submit-button"
>
  <span>
    Hello
  </span>
</button>
`;
